plugins/modules/bridge_vlan.py

```diff
@@ -0,0 +1,246 @@
+        _possible_vlans = out.strip().split(":")[1]
+
+        if rc != 0 or not _possible_vlans:
+            return []
+
+        # Output: bridge-port.vlans:120-129, 502 pvid untagged
+        # Output: bridge-port.vlans:120-129
+        # Output: bridge-port.vlans:120, 121, 122, 123
+        _vlans = _possible_vlans.split(",")
+        if "pvid" in _vlans[-1]:
+            # Remove the x pvid untagged which is the last
+            _vlans.pop()
+
+        return _vlans
+
+    def _apply_vlans(self, vlans: List[Any], taps: List[Any]) -> bool:
+        """Applies the provided VLANs for the given TAPs.
+
+        Args:
+            vlans (list) -> Containing the VLAN ids to be applied.
+            taps (list) -> Names of TAP devices to which VLANs are applied.
+
+        Return:
+            True if successful else False
+        """
+        _success = True
+        _base_cmd = [self.module.get_bin_path("bridge", True), "vlan", "add", "dev"]
+        for _tap in taps:
+            _tap_cmd = deepcopy(_base_cmd)
+            _tap_cmd.append(_tap)
+            _tap_cmd.append("vid")
+
+            for _vlan in vlans:
+                _vlan_cmd = deepcopy(_tap_cmd)
+                _vlan_cmd.append(_vlan)
+                (_rc, _, _) = self.execute_command(_vlan_cmd)
+
+                if _rc != 0 and _success:
+                    _success = False
+
+        return _success
+
+    def add_vlans(self) -> Dict[Any, Any]:
+        """Add the VLAN ids to the associated network."""
+        _result = dict(changed=False, failed=True, networks=dict())
+        for _net in self.vnet_names:
+            _rc, _out, _err = self._get_net_info(_net)
+            if _rc != 0:
+                _result["networks"][_net] = dict(failed=False, error=_err)
+                continue
+
+            net_info = json.loads(_out)
+            taps = get_taps(net_info)
+            if len(taps) == 0:
+                _result["networks"][_net] = dict(
+                    error="No TAP interfaces found.", failed=False
+                )
+                continue
+
+            iface = get_physical_iface(net_info)
+            if iface is None:
+                _result["networks"][_net] = dict(
+                    error="No physical interface found.", failed=False
+                )
+                continue
+
+            vlans = self._get_vlan_ids(iface)
+            if len(vlans) == 0:
+                _result["networks"][_net] = dict(
+                    error="Failed to gather the VLAN information.", failed=False
+                )
+                continue
+
+            if self._apply_vlans(vlans, taps):
+                _result["changed"] = True
+                _result["success"] = True
+                _result["failed"] = False
+                _result["networks"][_net] = deepcopy(net_info)
+            else:
+                _result["failed"] = True
+                _result["success"] = False
+                _result["networks"][_net] = deepcopy(net_info)
```

Comment:
I don't think this is OK. What happens if the the loop runs twice with the first iteration failing and the second one passing? The second iteration will set the success to True and failed to False, leading to an unexpected result.